packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,79 @@
+-   [Sectioned Combobox (issue 791)](https://github.com/ni/nimble/issues/791)
+-   [Prototype branch Storybook](https://60e89457a987cf003efc0a5b-cvmxlqwglb.chromatic.com/?path=/docs/components-select--docs) ([code](https://github.com/ni/nimble/tree/groupable-select))
+
+## Implementation / Design
+
+The HTML native `select` supports option groups with the native `optgroup` element. Configuring the `select` to use the `optgroup` takes the following form:
+
+```html
+<select name="cars">
+    <optgroup label="Swedish Cars">
+        <option value="volvo">Volvo</option>
+        <option value="saab">Saab</option>
+    </optgroup>
+    <optgroup label="German Cars">
+        <option value="mercedes">Mercedes</option>
+        <option value="audi">Audi</option>
+    </optgroup>
+</select>
+```
+
+The Nimble `Select` (and `Combobox`) will follow the same approach, and provide a new custom element, `nimble-list-option-group`, that mirrors the `optgroup` behavior/API:
+
+```html
+<nimble-select name="cars">
+    <nimble-list-option-group label="Swedish Cars">
+        <nimble-list-option value="volvo">Volvo</option>
+        <nimble-list-option value="saab">Saab</option>
+    </nimble-list-option-group>
+    <nimble-list-option-group label="German Cars">
+        <nimble-list-option value="mercedes">Mercedes</option>
+        <nimble-list-option value="audi">Audi</option>
+    </nimble-list-option-group>
+</nimble-select>
+```
+
+### API
+
+The `ListOptionGroup` class API should be fairly lightweight:
+
+```ts
+public class ListOptionGroup {
+    @attr
+    public label?: string;
+}
+```
+
+_Question: Should we support slotting icons (other content?) into the group display?_
+
+### Filtering
+
+As mentioned in the IxD, when a user has filtering enabled, typing into the filter input will match both individual options (as normal), as well as _all_ options that are under a group that matches the filter text.
+
+_Question: Should matching against a group be configurable?_
+
+#### Combobox Guidance
+
+It should be noted that when using groups with a `Combobox`, it is advised that each option still be unique (for both its display and value). This is predicated on the [decision](https://github.com/ni/nimble/issues/1168#issuecomment-1568733159) that the `Combobox` will change it's behavior for platforms like Angular, where the model value will _always_ be the text that is in the display. This will make it impossible to disambiguate between options that have the same display value under separate groups.
+
+## Alternative Implementations / Designs
+
+Originally, there was a desire to display the group name for the selected option outside of the dropdown of the component. However, there is not precedent for this for other design systems that support groups, nor the native `select`. If a client desires to display the group for the selected option that will be managed on the client side, including _where_ the display would be.
```

Comment:
Must the client have their own way of figuring out the group from the selected value (e.g. include the group label in the option value), or should we provide some API for getting the (selected item's) group label?

Reply:
My thought is that we should _not_ provide such an API. The native APIs offer no such thing (not that this is in and of itself a definitive reason to _not_ provide it, but it is a data point). I certainly wouldn't want an attribute API where a client is providing essentially redundant values for everything inside a group. This would then suggest a method the `Select` would provide to return the group for a particular option (as it would need to be `ListOption`-derived, and not value, since we provide no uniqueness guarantees for values), which is not a pattern we should promote.